src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -217,7 +217,10 @@ public void OnCompleted(Action continuation)
             else
             {
                 // THIS IS AN ERROR STATE - ONLY ONE WAITER CAN WAIT
-                throw new InvalidOperationException("Concurrent reads are not supported.");
```

Comment:
Right. @lodejard suggested putting all `await SocketInput`s behind a [TaskQueue](https://github.com/SignalR/SignalR/blob/3a219c101587333d562f04f30903d50ad9773e1c/src/Microsoft.AspNet.SignalR.Core/Infrastructure/TaskQueue.cs) to prevent the possibility of concurrent reads. Should I create an issue for that? It would be interesting to see how that affects performance.